projects/angular-auth-oidc-client/src/lib/api/data.service.ts

```diff
@@ -1,24 +1,37 @@
-import { HttpHeaders } from '@angular/common/http';
+import { HttpHeaders, HttpParams } from '@angular/common/http';
 import { Injectable } from '@angular/core';
 import { Observable } from 'rxjs';
+import { ConfigurationProvider } from '../config/config.provider';
 import { HttpBaseService } from './http-base.service';
 
 @Injectable()
 export class DataService {
-  constructor(private httpClient: HttpBaseService) {}
+  constructor(private httpClient: HttpBaseService, private readonly configurationProvider: ConfigurationProvider) {}
 
   get<T>(url: string, token?: string): Observable<T> {
     const headers = this.prepareHeaders(token);
+    let params = new HttpParams();
 
+    const { ngswBypass } = this.configurationProvider.getOpenIDConfiguration();
+    if (ngswBypass) {
+      params = params.set('ngsw-bypass', '');
+    }
     return this.httpClient.get<T>(url, {
       headers,
+      params,
     });
   }
 
   post<T>(url: string, body: any, headersParams?: HttpHeaders) {
     const headers = headersParams || this.prepareHeaders();
+    let params = new HttpParams();
+
+    const { ngswBypass } = this.configurationProvider.getOpenIDConfiguration();
+    if (ngswBypass) {
+      params = params.set('ngsw-bypass', '');
```

Comment:
If this is a custom header, we have to prefix it with an `x-` so the header would be `x-ngsw-bypass`

Reply:
Defining a constant for the header / param sounds good to me :+1: 

The first try of this feature was using the custom header `ngsw-byass`. The problem is that prefixing the header with `x-` will not work because of the following check in the service worker nsgw-worker.js:

```
onFetch(event) {
            ...
            if (req.headers.has('ngsw-bypass') || /[?&]ngsw-bypass(?:[=&]|$)/i.test(requestUrlObj.search)) {
                return;
            }
            ...
```
So the header must be exactly `ngsw-bypass`. Otherwise it will be ignored. Furthermore, I experienced the error that Keycloak rejects requests containing the custom header due to a CORS error because the custom header is not in the list of the allowed headers. I tried to add it somehow to keycloak but that was not easily possible and might also cause conflicts with other identity servers. 

Besides using a custom header, it's possible to use a **custom param** , which is what I would suggest. When using this approach a param will be appended to all auth-related requests: `https://identity-server.com/example?ngsw-bypass=`. and the CORS issues are gone.

I see that the description in the configuration.md is misleading and needs to be adjusted to the new approach. So my suggestion would be to define the constant `NGSW_CUSTOM_PARAM = ngsw-bypass` and update the description from `Adds the ngsw-bypass header to all requests` to `Adds the ngsw-bypass param to all requests`

Reply:
Sounds good to me. Please go with that. @damienbod ?